src/main/java/LottoApplication.java

```diff
@@ -0,0 +1,51 @@
+import domain.LottoMachine;
+import domain.Profit;
+import domain.WinningStatus;
+import domain.lastweekwinninglotto.LastWeekWinningBonusBall;
+import domain.lastweekwinninglotto.LastWeekWinningLotto;
+import domain.lottoticket.LottoTicket;
+import domain.lottoticket.Lottos;
+import domain.lottoticket.NumberOfLottoTicket;
+import domain.strategy.RandomLottoNumberStrategy;
+import ui.Printer;
+import ui.Receiver;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LottoApplication {
+    private final Printer printer = new Printer();
+    private final Receiver receiver = new Receiver();
+    private static LottoMachine lottoMachine = new LottoMachine();
+    private final RandomLottoNumberStrategy randomLottoNumberStrategy = new RandomLottoNumberStrategy();
+
+    public void run() {
+        printer.requestPurchaseAmount();
+        NumberOfLottoTicket numberOfLottoTicket = new NumberOfLottoTicket(receiver.receiveLottoTotalAmount());
+        Lottos lottos = makeLottos(numberOfLottoTicket);
+        printer.printAllLotto(lottos);
+        printer.requestLastWeekLottoWinningNumber();
+        List<Integer> LastWeekLottoWinningNumbers = receiver.receiveLastWeekLottoWinningNumbers();
+        LastWeekWinningLotto lastWeekWinningLotto = new LastWeekWinningLotto(LastWeekLottoWinningNumbers);
+        printer.requestLottoBonusBallNumber();
+        LastWeekWinningBonusBall lastWeekWinningBonusBall = new LastWeekWinningBonusBall(receiver.receiveLottoBonusBallNumber());
+        ArrayList<WinningStatus> getLottoPrices =  lottoMachine.Discriminator(lottos, lastWeekWinningLotto, lastWeekWinningBonusBall, numberOfLottoTicket);
+        Profit profit = new Profit(getLottoPrices);
+        printer.printLottoProfit(profit.getCalculatedProfit(numberOfLottoTicket));
```

Comment:
```suggestion
        List<Integer> LastWeekLottoWinningNumbers = receiver.receiveLastWeekLottoWinningNumbers();
        LastWeekWinningLotto lastWeekWinningLotto = new LastWeekWinningLotto(LastWeekLottoWinningNumbers);
        printer.requestLottoBonusBallNumber();
        LastWeekWinningBonusBall lastWeekWinningBonusBall = new LastWeekWinningBonusBall(receiver.receiveLottoBonusBallNumber());
        List<WinningStatus> getLottoPrices =  lottoMachine.Discriminator(lottos, lastWeekWinningLotto, lastWeekWinningBonusBall, numberOfLottoTicket);
        Profit profit = new Profit(getLottoPrices);
        printer.printLottoProfit(profit.getCalculatedProfit(numberOfLottoTicket));
```

일전에 리뷰받은 내용입니다!
자바의 유명한 규칙중엔 program to interfaces not implementations 라는 부분이 있습니다.


[객체 지향 기초 이야기 3, 유연함(Flexibility)](https://javacan.tistory.com/entry/OO-Basic-3-Flexibility)

위 글 참고해보시면 좋을것 같아요